pandas/core/index.py

```diff
@@ -973,7 +976,9 @@ def _convert_list_indexer(self, keyarr, kind=None):
         and we have a mixed index (e.g. number/labels). figure out
         the indexer. return None if we can't help
         """
-        if (kind is None or kind in ['iloc','ix']) and (is_integer_dtype(keyarr) and not self.is_floating()):
+        if (kind is None or kind in ['iloc', 'ix']) and (
+            is_integer_dtype(keyarr) and not self.is_floating() and not com.is_period_arraylike(keyarr)):
+
```

Comment:
on 2nd thought, this is going to cause a perf issue ,change this to ``isinstance(keyarr, ABCPeriodIndex)`` (you can import this at the top.